iModelCore/ECPresentation/Source/Serialization/DefaultECPresentationSerializer.cpp

```diff
@@ -738,7 +877,23 @@ NavNodeKeyPtr DefaultECPresentationSerializer::_GetBaseNavNodeKeyFromJson(BeJsCo
     {
     Utf8CP type = json["Type"].asCString();
     Utf8CP specificationIdentifier = json["SpecificationIdentifier"].asCString();
-    return NavNodeKey::Create(type, specificationIdentifier, ParseNodeKeyHashPath(json["PathFromRoot"]));
+    NavNodeKeyPtr key = NavNodeKey::Create(type, specificationIdentifier, ParseNodeKeyHashPath(json["PathFromRoot"]));
+    std::unique_ptr<PresentationQuery> instanceKeysSelectQuery = GetPresentationQueryFromJson(json["InstanceKeysSelectQuery"]);
+    if (!instanceKeysSelectQuery->GetQueryString().empty())
+        key->SetInstanceKeysSelectQuery(std::move(instanceKeysSelectQuery));

```

Comment:
I'd rather `GetPresentationQueryFromJson` didn't return invalid queries, in which case we could simply call
```cpp
key->SetInstanceKeysSelectQuery(GetPresentationQueryFromJson(json["InstanceKeysSelectQuery"]));
```

Reply:
Done.